### PR TITLE
Add editable dropdown

### PR DESCRIPTION
**Pull request by vlal**
* Editable dropdown, ability to add custom choices

### CODE REVIEW

**Comment by vlal** [comment]
@rndstr @jpellizzari is this mergeable?

File: src/components/EditableDropdown/EditableDropdown.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import Dropdown from '../Dropdown';
+
+
+const TextBox = styled.input`
+  margin-top: 2px;
+  padding: 6px;
+  font-size: 14px;
+  width: 240px;
+  outline-color: ${props => (props.changed ? 'orange' : 'default')};
```
Comment (jpellizzari) [inline src/components/EditableDropdown/EditableDropdown.js]: This needs to be a color from our theme not a hard-coded string. `props.theme.colors.status.error` would be my vote.

File: src/components/EditableDropdown/EditableDropdown.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  handleInputChange(event) {
+    this.setState({
+      currentValue: event.target.label,
+    });
+  }
+
+  render () {
+    return (
+      <div style={{position: "relative"}}>
```
Comment (jpellizzari) [inline src/components/EditableDropdown/EditableDropdown.js]: No inline styles please

File: src/components/EditableDropdown/EditableDropdown.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+`;
+
+class EditableDropdown extends React.Component {
+  constructor(props, context) {
+    super(props, context)
+
+    this.dropdown = null;
+
+    this.state = {
+      currentValue: "",
```
Comment (jpellizzari) [inline src/components/EditableDropdown/EditableDropdown.js]: Single-quotes here. Our linter will probably complain about this.

File: src/components/EditableDropdown/EditableDropdown.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    this.dropdown = ref;
+  }
+
+  handleChange(ev, value, label) {
+    if (this.props.onChange) {
+      this.props.onChange(ev, value, label);
+    }
+
+    this.setState({
+      currentValue: label,
```
Comment (jpellizzari) [inline src/components/EditableDropdown/EditableDropdown.js]: This should be the `value`

File: src/components/EditableDropdown/EditableDropdown.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    this.setState({
+      currentValue: event.target.label,
+    });
+  }
+
+  render () {
+    return (
+      <div style={{position: "relative"}}>
+        <TextBox type="text" onFocus={this.handleFocus} value={this.state.currentValue} onChange={this.handleInputChange} placeholder={this.props.placeholder} />
+        <Dropdown className={this.props.className} innerRef={this.setDropdownRef} onChange={this.handleChange} items={this.props.items}/>
```
Comment (jpellizzari) [inline src/components/EditableDropdown/EditableDropdown.js]: Why do we need to save the `Dropdown` ref? I don't see it used, but there may be something I'm not seeing.

**Pull request closed without merging**